org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/javaeditor/SemanticHighlightingReconciler.java

```diff
@@ -484,6 +514,96 @@ private void reconcilePositions(ASTNode[] subtrees) {
+				case BRACKET:
+					return findEditorHighlighting(PreferenceConstants.EDITOR_JAVA_BRACKET_COLOR);
+				case MULTI_LINE_COMMENT:
+					return findEditorHighlighting(PreferenceConstants.EDITOR_MULTI_LINE_COMMENT_COLOR);
+				case STRING:
+					return findEditorHighlighting(PreferenceConstants.EDITOR_STRING_COLOR);
+				case METHOD:
+					return findHighlighting(SemanticHighlightings.METHOD);
+				case ABSTRACT_CLASS:
+					return findHighlighting(SemanticHighlightings.ABSTRACT_CLASS);
+				case ABSTRACT_METHOD_INVOCATION:
+					return findHighlighting(SemanticHighlightings.ABSTRACT_METHOD_INVOCATION);
+				case ANNOTATION:
+					return findHighlighting(SemanticHighlightings.ANNOTATION);
+				case ANNOTATION_ELEMENT_REFERENCE:
+					return findHighlighting(SemanticHighlightings.ANNOTATION_ELEMENT_REFERENCE);
+				case AUTOBOXING:
+					return findHighlighting(SemanticHighlightings.AUTOBOXING);
+				case CLASS:
+					return findHighlighting(SemanticHighlightings.CLASS);
+				case DEPRECATED_MEMBER:
+					return findHighlighting(SemanticHighlightings.DEPRECATED_MEMBER);
+				case ENUM:
+					return findHighlighting(SemanticHighlightings.ENUM);
+				case FIELD:
+					return findHighlighting(SemanticHighlightings.FIELD);
+				case INHERITED_FIELD:
+					return findHighlighting(SemanticHighlightings.INHERITED_FIELD);
+				case INHERITED_METHOD_INVOCATION:
+					return findHighlighting(SemanticHighlightings.INHERITED_METHOD_INVOCATION);
+				case INTERFACE:
+					return findHighlighting(SemanticHighlightings.INTERFACE);
+				case LOCAL_VARIABLE:
+					return findHighlighting(SemanticHighlightings.LOCAL_VARIABLE);
+				case LOCAL_VARIABLE_DECLARATION:
+					return findHighlighting(SemanticHighlightings.LOCAL_VARIABLE_DECLARATION);
+				case METHOD_DECLARATION:
+					return findHighlighting(SemanticHighlightings.METHOD_DECLARATION);
+				case NUMBER:
+					return findHighlighting(SemanticHighlightings.NUMBER);
+				case PARAMETER_VARIABLE:
+					return findHighlighting(SemanticHighlightings.PARAMETER_VARIABLE);
+				case STATIC_FIELD:
+					return findHighlighting(SemanticHighlightings.STATIC_FIELD);
+				case STATIC_FINAL_FIELD:
+					return findHighlighting(SemanticHighlightings.STATIC_FINAL_FIELD);
+				case STATIC_METHOD_INVOCATION:
+					return findHighlighting(SemanticHighlightings.STATIC_METHOD_INVOCATION);
+				case TYPE_ARGUMENT:
+					return findHighlighting(SemanticHighlightings.TYPE_ARGUMENT);
+				case TYPE_VARIABLE:
+					return findHighlighting(SemanticHighlightings.TYPE_VARIABLE);
+				case RESTRICTED_IDENTIFIER:
+					return findHighlighting(SemanticHighlightingsCore.RESTRICTED_KEYWORDS);
+				case DEFAULT:
+				default:
+					return findEditorHighlighting(PreferenceConstants.EDITOR_JAVA_DEFAULT_COLOR);
+			}
+		}
+		return null;
+	}
+
+	private Highlighting findHighlighting(String preferenceKey) {
+		for (int i = 0; i < fJobSemanticHighlightings.length; i++) {
```

Comment:
this codes looks like fJobSemanticHighlightings and fJobHighlightings needs to be of same size, which is documented no where. You may even better use a single Collection containing tuples to make that clear.

Reply:
I have added `String fKey` to `Highlighting` class definition thus we won't have to look outside the `Highlighting[]`